### DISCUSSION TITLE
Documenting `telemetry` events with `telemetry_registry`

### DISCUSSION THREAD

**Discussion by isaacsanders** [quantum-elixir/quantum-core#491] [Ideas]
This would enable integration with tools like https://github.com/opentelemetry-beam/opentelemetry_telemetry

I'm also happy to do the work.

**Comment by maennchen** [comment]
@isaacsanders I haven't really looked at that so far.

Can you tell me what the advantages are and how this would be used?

Does this add a dependency?

**Reply by isaacsanders** [reply]
It automates the detection of emitted events. It doesn’t affect the runtime in any way.

it adds a dependency.

**Reply by maennchen** [reply]
I'm generally a bit hesitant to add new dependencies.

I think the library looks well-maintained though and is actually part of `beam-telemetry`.

I think this would be a good addition. A PR would be very welcome.

**Reply by isaacsanders** [reply]
I appreciate your hesitation. I should be able to put something together today or Monday.

**Reply by maennchen** [reply]
Moved to issue #492
